tests/components/denonavr/test_media_player.py

```diff
@@ -0,0 +1,57 @@
+"""The tests for the denonavr media player platform."""
+from unittest.mock import patch
+
+import pytest
+
+from homeassistant.components import media_player
+from homeassistant.components.denonavr import ATTR_COMMAND, DOMAIN, SERVICE_GET_COMMAND
+from homeassistant.const import ATTR_ENTITY_ID, CONF_HOST, CONF_NAME, CONF_PLATFORM
+from homeassistant.setup import async_setup_component
+
+NAME = "fake"
+ENTITY_ID = f"{media_player.DOMAIN}.{NAME}"
+
+
+@pytest.fixture(name="client")
+def client_fixture():
+    """Patch of client library for tests."""
+    with patch(
+        "homeassistant.components.denonavr.media_player.denonavr.DenonAVR",
+        autospec=True,
+    ) as mock_client_class, patch(
+        "homeassistant.components.denonavr.media_player.denonavr.discover"
+    ):
+        mock_client_class.name = NAME
+        mock_client_class.return_value.zones = {"Main": mock_client_class}
```

Comment:
It should be the instance of the client class that is stored in the zones dictionary not the class. So put the mock class return_value there instead.

https://github.com/scarface-4711/denonavr/blob/953cb68408ecebf90d8b0619d72bba6e5342f7fe/denonavr/denonavr.py#L243